stacker_blueprints/dynamodb.py

```diff
@@ -0,0 +1,162 @@
+from stacker.blueprints.base import Blueprint
+
+from troposphere import (
+    dynamodb2,
+    Ref,
+    GetAtt,
+    Output,
+)
+
+from troposphere.route53 import RecordSetType
+from . import util
+
+MAX_GSI_VALUE = 5
+
+
+def prep_schemata(config):
+    try:
+        schemata = []
+        for schema in config["KeySchema"]:
+            schemata.append(dynamodb2.KeySchema(**schema))
+
+    except KeyError:
+        raise KeyError(
+            "The key schema is required for the creation of a DynamoDB table."
+        )
+
+    return schemata
+
+
+def prep_projection(config):
+    return dynamodb2.Projection(**config["Projection"])
+
+
+def prep_throughput(config):
+    try:
+        return dynamodb2.ProvisionedThroughput(
+            **config["ProvisionedThroughput"]
+        )
+    except KeyError:
+        raise KeyError(
+            "ProvisionedThroughput values are required for the creation of " +
+            "a DynamoDB table or index."
+        )
+
+
+def prep_config(raw_config):
+    prepped_config = {}
+
+    dynamodb_table_properties = [
+        "AttributeDefinitions",
+        "GlobalSecondaryIndexes",
+        "KeySchema",
+        "LocalSecondaryIndexes",
+        "ProvisionedThroughput",
+        "StreamSpecification",
+    ]
+
+    util.check_properties(
+        raw_config,
+        dynamodb_table_properties,
+        "DynamoDB",
+    )
+
+    # AttributeDefinitions are required, so raise an IndexError if this
+    # doesn't work.
+    try:
+        attributes = []
+        for attribute in raw_config["AttributeDefinitions"]:
```

Comment:
same thing as @phobologic comment above re: small as possible section for wrapped errors